dashboard/app/controllers/application_controller.rb

```diff
@@ -351,7 +351,7 @@ def prevent_caching
       users_set_student_information_path,
     ].include?(request.path)
 
-    redirect_to lockout_path if Policies::ChildAccount.locked_out?(current_user)
+    redirect_to lockout_path if Policies::ChildAccount::ComplianceState.locked_out?(current_user)
```

Comment:
What do you think about creating a method like `Services::ChildAccount.refresh` where all our time_based transition logic is contained? Otherwise all the conditional transition logic will end up living here, in the ApplicationController, for example if the user should be transitions from `nil` -> `grace_period`. We could even add a transition to from [grace_period, locked_out] -> permission_granted. Then all our unit tests can be in one place and the rest of the code base can simply call `refresh` and assume the user in in the correct CPA state.

By the way, I'm not asking for you to add all the other transition to `refresh`, yet. I'm only asking you to consider starting the pattern so we can follow it with the other transitions.